composer/callbacks/system_metrics_monitor.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""System metrics monitor callback."""
+
+from __future__ import annotations
+
+import logging
+
+import psutil
+from py3nvml import py3nvml
+import os
+
+from composer.core import Callback, State, Event
+from composer.loggers import Logger
+
+log = logging.getLogger(__name__)
+
+__all__ = ["SystemMetricsMonitor"]
+
+
+class SystemMetricsMonitor(Callback):
+    """Track system metrics."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        py3nvml.nvmlInit()
+
+    def run_event(self, event: Event, state: State, logger: Logger):
+        # skip the microbatch events
+        if event in [
+            Event.BEFORE_FORWARD,
+            Event.AFTER_FORWARD,
+            Event.BEFORE_LOSS,
+            Event.AFTER_LOSS,
+            Event.BEFORE_BACKWARD,
+            Event.AFTER_BACKWARD,
+        ]:
+            return
+        system_metrics = self.compute_system_metrics()
```

Comment:
you could do something like this to gather all nodes' system metrics
```suggestion
        local_node_system_metrics = self.compute_system_metrics()
        all_system_metrics = dist.all_gather_object(local_node_system_metrics)
        system_metrics = {key: value for local_metrics  in all_system_metrics for key, value in local_metrics.items()}
```